tools/add_triage_signature.py

```diff
@@ -2105,6 +2105,66 @@ def _process_ticket(self, url, issue_key):
         self._update_triaging_ticket("\n".join(known_tags_messages + unknown_tag_message))
 
 
+class FailedRequestTriggersHostTimeout(Signature):
+    """
+    This signature looks for failed requests that could have caused an host timeout, ultimately failing installation
+    """
+
+    # This has to be maintained to be the same format as
+    # https://github.com/openshift/assisted-installer-agent/blob/aebd94105b4ed6442f21a7a26ab4e40eafd936aa/src/commands/step_processor.go#L81
+    # Remember to maintain backwards compatibility if that format ever changes - create
+    # PATTERN_NEW and try to match both.
+    LOG_PATTERN = re.compile(
+        r'time="(?P<time>.+)" level=(?P<severity>[a-z]+) msg="(?P<message>.*api\.openshift\.com/api/assisted-install.*Service Unavailable)" file=.+'
+    )
+
+    HOST_TIMED_OUT_STATUS_INFO = "Host failed to install due to timeout while connecting to host"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            comment_identifying_string="h1. Failed request triggering host timeout",
+        )
+
+    def _process_ticket(self, url, issue_key):
+        report = ""
+
+        cluster = get_metadata_json(url)["cluster"]
+        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster["id"])
+
+        failed_requests_hosts = self._failed_requests_hosts(cluster, triage_logs_tar)
+        timed_out_hosts = self._timed_out_hosts(cluster)
+
+        for host_id in timed_out_hosts & failed_requests_hosts:
+            report += f"h2. Host {host_id} has request failures and timed out. Did the request cause the host to timeout?"
```

Comment:
The reason I used an array rather than a string is so that I could "\n" join it at the end to handle newlines automatically, any reason you changed it back to string?

Reply:
Yeah my bad, got carried away when changing the message :facepalm: